pkg/tbtc/redemption.go

```diff
@@ -0,0 +1,201 @@
+package tbtc
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"time"
+
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/chain"
+)
+
+// RedemptionTransactionShape is an enum describing the shape of
+// a Bitcoin redemption transaction.
+type RedemptionTransactionShape uint8
+
+const (
+	// RedemptionChangeFirst is a shape where the change output is the first one
+	// in the transaction output vector. This shape makes the change's position
+	// fixed and leverages some SPV proof cost optimizations made in the Bridge
+	// implementation.
+	RedemptionChangeFirst RedemptionTransactionShape = iota
+	// RedemptionChangeLast is a shape where the change output is the last one
+	// in the transaction output vector.
+	RedemptionChangeLast
+)
+
+// RedemptionRequest represents a tBTC redemption request.
+type RedemptionRequest struct {
```

Comment:
Some of the fields, for example `TxMaxFee` and `RequestedAt`, are currently unused but I take the plan was to reflect the [on-chain structure](https://github.com/keep-network/tbtc-v2/blob/66a04c618e140230d25ea0d784ed75e6ba3030a2/solidity/contracts/bridge/Redemption.sol#L137) 1:1 because we'll use those fields in the redemption action anyway. Is this correct assumption?

Reply:
Yeah, correct.